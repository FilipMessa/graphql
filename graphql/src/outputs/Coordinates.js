// @flow

import { GraphQLObjectType, GraphQLFloat } from 'graphql';

import type { Coordinates } from '../types/Location';

export default new GraphQLObjectType({
  name: 'Coordinates',
  fields: {
    lat: {
      type: GraphQLFloat,
      description: 'Latitude.',
      resolve: ({ lat }: Coordinates): number => lat,
    },

    lng: {
      type: GraphQLFloat,
      description: 'Longitude.',
      resolve: ({ lng }: Coordinates): number => lng,
    },
  },
});
