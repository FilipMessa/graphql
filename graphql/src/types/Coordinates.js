// @flow

import { GraphQLObjectType, GraphQLFloat } from 'graphql';

import type { CoordinatesType } from '../Entities';

export default new GraphQLObjectType({
  name: 'Coordinates',
  fields: {
    lat: {
      type: GraphQLFloat,
      resolve: ({ latitude }: CoordinatesType): number => latitude,
    },

    lng: {
      type: GraphQLFloat,
      resolve: ({ longitude }: CoordinatesType): number => longitude,
    },
  },
});
