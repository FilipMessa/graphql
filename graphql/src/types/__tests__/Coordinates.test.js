// @flow

import Coordinates from '../Coordinates';

it('Coordinates type should have valid fields', () => {
  expect(Coordinates.getFields()).toMatchSnapshot();
});
