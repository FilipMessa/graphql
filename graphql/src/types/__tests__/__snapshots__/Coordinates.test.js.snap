// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coordinates have valid fields 1`] = `
Object {
  "lat": Object {
    "args": Array [],
    "isDeprecated": false,
    "name": "lat",
    "resolve": [Function],
    "type": "Float",
  },
  "lng": Object {
    "args": Array [],
    "isDeprecated": false,
    "name": "lng",
    "resolve": [Function],
    "type": "Float",
  },
}
`;
